fpdf/fpdf.py

```diff
@@ -2156,29 +2199,39 @@ def _render_styled_cell_text(
                     f"{(x + w) * k:.2f} {(self.h - (y + h)) * k:.2f} l S "
                 )
 
-        if styled_txt_frags:
+        s_start = self.x
+        s_width, underlines = 0, []
+        if text_line.fragments:
             if align == "R":
                 dx = w - self.c_margin - styled_txt_width
             elif align == "C":
                 dx = (w - styled_txt_width) / 2
             else:
                 dx = self.c_margin
+            s_start += dx
 
             if self.fill_color != self.text_color:
                 s += f"q {self.text_color} "
 
             prev_font_style, prev_underline = self.font_style, self.underline
-            s_width, underlines = 0, []
             s += (
                 f"BT {(self.x + dx) * k:.2f} "
                 f"{(self.h - self.y - 0.5 * h - 0.3 * self.font_size) * k:.2f} Td"
             )
-            # If multibyte, Tw has no effect - do word spacing using an
-            # adjustment before each space
-            if self.ws and self.unifontsubset:
+
+            word_spacing = 0
+            if align == "J" and text_line.number_of_spaces_between_words:
+                word_spacing = (
+                    w - self.c_margin - self.c_margin - styled_txt_width
+                ) / text_line.number_of_spaces_between_words
+            if word_spacing and self.unifontsubset:
+                # If multibyte, Tw has no effect - do word spacing using an
+                # adjustment before each space
                 space = escape_parens(" ".encode("utf-16-be").decode("latin-1"))
-                s += " 0 Tw"
-                for frag in styled_txt_frags:
+                if self.ws > 0:
```

Comment:
This is not covered by tests: in what case is it needed?

Reply:
It will be used when on the same page first some justified text with a non-unicode font is printed (leaving behind a non-0 `self.ws`), and after that some other text with an unicode font. I guess we don't have any tests mixing font types this way.